src/System.Private.Xml/src/System/Xml/XPath/XPathNodeIterator.cs

```diff
@@ -97,35 +96,5 @@ public virtual void Reset()
                 _iterationStarted = false;
             }
         }
-
-        private struct DebuggerDisplayProxy
```

Comment:
I think we should check if this class is working correctly under the debugger, if not we might be missing https://msdn.microsoft.com/en-us/library/system.diagnostics.debuggerdisplayattribute(v=vs.110).aspx and use this class to display

Reply:
Time for an embarrassing newb question: How do I find the correct DLL to experiment with? I looked for recently updated `*xml.dll` files, but the one that made the most sense (\corefx\bin\runtime\netcoreapp-Windows_NT-Debug-x64), gave me an error ("The reference is invalid or unsupported") when I tried to load it into a core project.

Reply:
@krwq for q above.

Reply:
@MisinformedDNA the easy way to check this would be this:
- build corefx repo
- rebuild System.Private.Xml (`msbuild /t:Rebuild`)
- run tests in any of the xml projects using your code path (`msbuild /t:RebuildAndTest`)
- open sln file from root of System.Private.Xml directory
- right click on the test project you have previously built and "Set as StartUp project"
- now you should be able to "Start" running tests from that project and be able to use debugger
- to run a single test you can change command line used in the project properties (right click on the test project and Debug tab) and do it according with Xunit runner instructions
- one way you can do above is put `[Trait("foo", "bar")]` on your test and in the command line add following arg: `-trait foo=bar` - this will cause to run only tests with this custom attribute
- after you set a breakpoint DebuggerTypeProxyAttribute should influence how the data is being shown under the debugger - please refer to MSDN page and some articles about it for more details

Reply:
@MisinformedDNA to compare to the expected behavior simply create full framework app in VS using one of these classes and see if it shows the same under the debugger

Reply:
Class looks under the debugger. Full framework and newest code match up.